### ISSUE TITLE
Could not find com.czt.mp3recorder:library:1.0.4.

### ISSUE THREAD

**Issue by nitinmmnova** [amorenew/record_mp3_plus#2]
Everything works fine and I just update android studio and it's not working I tried so many things but no any luck

How I can fix this error?

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkDebugAarMetadata'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.czt.mp3recorder:library:1.0.4.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/czt/mp3recorder/library/1.0.4/library-1.0.4.pom
       - https://repo.maven.apache.org/maven2/com/czt/mp3recorder/library/1.0.4/library-1.0.4.pom
       - https://jitpack.io/com/czt/mp3recorder/library/1.0.4/library-1.0.4.pom
       - https://storage.googleapis.com/download.flutter.io/com/czt/mp3recorder/library/1.0.4/library-1.0.4.pom
       - https://jcenter.bintray.com/com/czt/mp3recorder/library/1.0.4/library-1.0.4.pom
       - https://www.jitpack.io/com/czt/mp3recorder/library/1.0.4/library-1.0.4.pom
     Required by:
         project :app
   > Could not find com.czt.mp3recorder:library:1.0.4.

**Comment by The-EDev** [comment]
This is due to JCenter shutting down, this library doesn't exist in Maven Central (which JCenter now redirects to), and attempting to compile it manually requires a huge amount of outdated dependencies, many of which also don't exist in any.

TL;DR: the library's dependencies are completely broken and there's pretty much no solution :(

**Comment by ludy-auror** [comment]
I've done some research, I agree with @The-EDev about the root cause. As to how the fix, there are basically 2 approaches:

- Upload the AAR library `com.czt.mp3recorder:library` into the new mavenCentral repository. This is the best in terms of ease of use
- Import the AAR library locally

**Comment by amorenew** [comment]
@ludy-auror I tried to download the aar file from
https://mvnrepository.com/artifact/com.czt.mp3recorder/library/1.0.4
but I can't download it
<img width="710" alt="image" src="https://github.com/user-attachments/assets/23d19f69-394b-456e-a8af-186393c6c54b">

and I tried to compile the original repo
https://github.com/GavinCT/AndroidMP3Recorder
but it failed to sync with the gradle

**Comment by amorenew** [comment]
it is possible to make a new library project and then build it into an aar library
for
https://github.com/GavinCT/AndroidMP3Recorder
but this would take 2 to 3 hours, so I need help with that
my mac is an intel mac so it is slow, guys with M mac can make it faster

**Comment by The-EDev** [comment]
> I tried to download the aar file from
> https://mvnrepository.com/artifact/com.czt.mp3recorder/library/1.0.4
> but I can't download it

This is because the spring repository isn't public (not anymore at least)



> it is possible to make a new library project and then build it into an aar library

I personally tried to do this, but ended up giving up because of how many outdated libraries it requires (most of which need to be compiled themselves and have not received any security updates in over 10 years).

**Comment by amorenew** [comment]
it is possible but it is time-consuming 
 it will require a new library project then we add `jni`, `jniLibs` and `Java folder ` to the new library project and export aar file then we test the mp3 recorder with the new aar file that's why it is a time consuming task

**Comment by ludy-auror** [comment]
@amorenew @The-EDev Luckily I have the AAR file in my local gradle cache, so no compile needed
I'll get a PR up using the second approach soon. 

> Import the AAR library locally

Also if any of you could upload the AAR file to a maven repository, you can get the file from my PR, that will make things much easier for everyone

**Comment by LudySu** [comment]
Hi guys, bad thing I've been using an incorrect Github account.
Now I've created a PR that imports the AAR library locally, since I still have a copy of the original AAR artifact in my local build cache

**Comment by amorenew** [comment]
@LudySu I added the whole library files to my project so no need for the aar file or the extra step to download it

https://github.com/amorenew/record_mp3_plus/pull/4

Could someone test it and let me know if it works as expected?

```
  record_mp3_plus:
    git:
      url: git@github.com:amorenew/record_mp3_plus.git
      ref: fix/gradle-build

```

**Comment by ludy-auror** [comment]
@amorenew Great, it works on my machine too!

**Comment by LudySu** [comment]
@amorenew Great, it works on my machine too!

**Comment by amorenew** [comment]
I published a new release with the fix

**Comment by nitinmmnova** [comment]
Thanks!!
